DMTN-148.tex

```diff
@@ -50,9 +50,255 @@
+\verb|obs_CAMERA_data| package implement one-off mapper methods to allow the
+product to be found.  However, only \verb|DEFECTS|, \verb|QE_CURVE|, and \verb|LINEARITY|
+follow this system, leaving the remaining products to be handled with
+camera-specific code and cameraGeom entries.  In addition to having
+diverse implementations, these handlers are often rigid, and cannot be
+versioned easily.
+
+\section{Transitioning to Gen3/What We Should Expect From Gen3}
+
+\subsection{A CalibrationBaseClass}
+The first step to a consistent calibration system is to ensure that
+all products in both generations use the \verb|DEFECTS|-like calibration
+product system.  For FITS image products, this requires future
+calibrations to be constructed with all of the header keyword values
+listed above.
+
+The remaining unconverted calibration types (\verb|CROSSTALK|, \verb|BFKERNEL|)
+should be converted to use \verb|DEFECTS|-like class, most likely by defining
+a CalibrationBaseClass, and making all of the calibration products
+subclasses of that.  This would enforce that all calibrations can be
+relied upon to provide getMetadata, setMetadata, readFits, writeFits,
+readText, and writeText methods.  Gen2 Butler instances would then be
+able to access all calibration products in a single consistent manner.
+This also ensures that Gen2 and Gen3 can use the same calibration
+products.
+
+\subsection{Calibration Construction in Gen3}
+
+Calibration products should be constructed using well defined
+PipelineTask code.  This removes the need to delineate between
+``calibrations'' and ``user-curated calibrations''.  The only
+remaining difference between these two groups is what the initial
+persisted data type is: ``calibrations'' are stored in FITS images, as
+they represent corrections that are only meaningful as full images;
+``user-curated calibrations'' can be visualized in a small number of
+parameters, and so persisting these in a text format to allow user
+inspection is reasonable.  ECSV and YAML are good formats for these
+calibrations, and a calibration's readText/writeText methods should
+implement one of these.
+
+In addition to the calibration product, the PipelineTask code must
+persist a record of the input data used for construction.  This, along
+with the already persisted pipeline definition and task configuration,
+allows the calibration to be regenerated by any user, making the
+construction process much more transparent than it currently is.  A
+YAML file containing the dataId parameters for all inputs would
+satisfy this.  Appending additional construction statistics (for
+example, weight factors applied to each input during the combination
+stage) would be helpful in the validation process.
+
+Newly generated calibration products will belong to a Gen3 collection,
+and that collection will be used to manage the calibration through the
+validation and certification process.  This collection may only
+contain one entry for a given OIDF set.
```

Comment:
OIDF? `OBSTYPE`/`INSTRUME`/`DETECTOR`/`FILTER`? Can we spell it out?

Reply:
Oh, it is in `myacronyms.txt`, but the PDF I'm looking at doesn't include a glossary!

Reply:
It took me a bit of a moment to work out what that was as well but my PDF did have it at the end.

Reply:
I must have been confused by the dev directions.  I'll sort out why it doesn't appear.